porch/engine/pkg/kpt/render_test.go

```diff
@@ -0,0 +1,192 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kpt
+
+import (
+	"testing"
+
+	"github.com/GoogleContainerTools/kpt/internal/printer/fake"
+	"github.com/GoogleContainerTools/kpt/internal/util/render"
+	"github.com/google/go-cmp/cmp"
+)
+
+const (
+	simpleBucketBucket = `
+apiVersion: storage.cnrm.cloud.google.com/v1beta1
+kind: StorageBucket
+metadata: # kpt-merge: config-control/blueprints-project-bucket
+  name: blueprints-project-bucket # kpt-set: ${project-id}-${name}
+  namespace: config-control # kpt-set: ${namespace}
+  annotations:
+    cnrm.cloud.google.com/force-destroy: "false"
+    cnrm.cloud.google.com/project-id: blueprints-project # kpt-set: ${project-id}
+spec:
+  storageClass: standard # kpt-set: ${storage-class}
+  uniformBucketLevelAccess: true
+  versioning:
+    enabled: false
+`
+
+	simpleBucketKptfile = `
+apiVersion: kpt.dev/v1
+kind: Kptfile
+metadata:
+  name: simple-bucket
+  annotations:
+    blueprints.cloud.google.com/title: Google Cloud Storage Bucket blueprint
+info:
+  description: A Google Cloud Storage bucket
+pipeline:
+  mutators:
+    - image: gcr.io/kpt-fn/apply-setters:v0.2.0
+      configPath: setters.yaml
+`
+
+	simpleBucketSetters = `
+apiVersion: v1
+kind: ConfigMap
+metadata: # kpt-merge: /setters
+  name: setters
+data:
+  name: updated-bucket-name
+  namespace: updated-namespace
+  project-id: updated-project-id
+  storage-class: updated-storage-class
+`
+)
+
+func TestRenderWithFunctionConfigFile(t *testing.T) {
+	t.Skip("kpt renderer does not correctly construct function config")
```

Comment:
@phanimarupaka for now I am skipping this test; when we fix the renderer handling of configPath we can remove this line.